pkg/sign/certificate.go

```diff
@@ -220,3 +241,27 @@ func (f *Fulcio) GetCertificate(ctx context.Context, keypair Keypair, opts *Cert
 
 	return certBlock.Bytes, nil
 }
+
+// subjectFromClaims returns the claim to use for proof of key possession
+// signature.
+//
+// FIXME: this is taken from sigstore/sigstore. We don't currently use the
+// exported SubjectFromToken function because it requires an *oidc.IDToken
+// object, which is generated after the token is verified by the client.
+// In this library, the token is not verified as Fulcio also verifies the
+// token's signature anyway. Also, sigstore/sigstore's implementation of
+// SubjectFromToken operates on the token's bytes (tok.Claims() wraps around
+// json.Unmarshal), so possibly this change belongs there.
+func subjectFromClaims(c jsonWebToken) (string, error) {
+	if c.Email != "" {
+		if !c.Verified {
+			return "", errors.New("not verified by identity provider")
```

Comment:
Would it make sense to fall through to check `c.Sub` in this case?

Reply:
What are the scenarios where an ID Token contains an (unverified) email but we wish to use the subject instead? I feel like require email + verified for email, but fail if only email is provided would result in a more consistent behaviour with less surprises? I.e I have a slight preference to keep as is.